pinot-common/src/main/java/org/apache/pinot/common/function/scalar/ArrayFunctions.java

```diff
@@ -136,4 +136,13 @@ public static int[] arrayConcatInt(int[] values1, int[] values2) {
   public static String[] arrayConcatString(String[] values1, String[] values2) {
     return ArrayUtils.addAll(values1, values2);
   }
+
+  @ScalarFunction
+  public static double arraySum(Double[] values) {
```

Comment:
Currently we don't support having different parameter type for the same scalar function, so we might need to use the same workaround similar to other functions by adding the type into the function name